### PR TITLE
Add debug logs that mgiht help with lockups

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

File: src/core/state.go [hunk 1]
```diff
@@ -1136,7 +1150,7 @@ func (state *BuildState) queueTargetAsync(target *BuildTarget, forceBuild, build
 		// Wait for these targets to actually build.
 		if building {
 			for _, t := range target.Dependencies() {
-				t.WaitForBuild()
+				t.WaitForBuild(target.Label)
```
Comment (samwestmoreland) [inline src/core/state.go:1153]: What's the reason for this change? (Just cuz it sticks out as not just a debugging change)

**Review by samwestmoreland — APPROVED** [review]

File: src/core/state.go (hunk 1 above)
Comment (Tatskaari) [reply src/core/state.go:1153]: I need the dependency information so I can add it to the log message
